client/allocrunner/cni/plugins.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package cni
+
+// Generic has the one key that all plugins must have: "type"
+type Generic struct {
+	Type string `json:"type"`
+}
+
+// Bridge is the subset of options that we use to configure the "bridge" plugin.
+// https://www.cni.dev/plugins/current/main/bridge/
+type Bridge struct {
+	Type         string `json:"type"`
+	Bridgename   string `json:"bridge"`
+	IpMasq       bool   `json:"ipMasq"`
+	IsGateway    bool   `json:"isGateway"`
+	ForceAddress bool   `json:"forceAddress"`
+	HairpinMode  bool   `json:"hairpinMode"`
+	Ipam         IPAM   `json:"ipam"`
+}
+type IPAM struct {
+	Type   string    `json:"type"`
+	Ranges [][]Range `json:"ranges"`
+	Routes []Route   `json:"routes"`
+}
+type Range struct {
+	Subnet string `json:"subnet"`
+}
+type Route struct {
+	Dst string `json:"dst"`
+}
+
+// Firewall is the "firewall" plugin.
+// https://www.cni.dev/plugins/current/meta/firewall/
+type Firewall struct {
+	Type           string `json:"type"`
+	Backend        string `json:"backend"`
+	AdminChainName string `json:"iptablesAdminChainName"`
+}
+
+// Portmap is the "portmap" plugin.
+// https://www.cni.dev/plugins/current/meta/portmap/
+type Portmap struct {
+	Type         string         `json:"type"`
+	Capabilities CapabilityArgs `json:"capabilities"`
+	Snat         bool           `json:"snat"`
+}
+type CapabilityArgs struct {
```

Comment:
We have to export these types because of Go's rules around serialization, which means the type is visible outside this package. Given this struct belongs only to the `Portmap` plugin, might it be better named something like `PortmapCapabilities`?